Week 5/PalindromeLinkedList.cpp

```diff
@@ -0,0 +1,66 @@
+#include <iostream>
+
+using namespace std;
+
+struct ListNode{
+    int val;
+    ListNode *next;
+    ListNode(int x) : val(x), next(NULL) {};
+};
+
+// Reverse function
+ListNode* reverse(ListNode *head) {
+    ListNode *prev = NULL;
+    ListNode *curr = head;
+    ListNode *nxt;
+    while(curr != NULL) {
+        nxt = curr->next;
+        curr->next = prev;
+        prev = curr;
+        curr = nxt;
+    }
+    
+    return prev;
+}
+
+bool isPalindrome(ListNode *head) {
+    // Check for NULL
+    if(head != NULL || head->next != NULL) {
```

Comment:
I think this condition in if statement should be the opposite. Also, I think its totally valid if we receive a linked list with just one element, so we dont have to throw error when head.next is null.

So ideally, the code for input validation comes down to:
if (head == null )
    throw exception.